src/js/components/DateInput/DateInput.js

```diff
@@ -177,6 +177,9 @@ const DateInput = forwardRef(
               setOpen(true);
               if (onFocus) onFocus(event);
             }}
+            onMouseDown={({ target }) => {
```

Comment:
Is it the kinda behavior we're able to cover with tests?

Reply:
Yes, I think so

Reply:
Typically, we try to avoid `onMouseDown` to achieve a click behavior because a user may mouse down and drag away from the input and decide not to click. Is there a reason we can't do this with `onClick` handler?

Reply:
I've tried implement the solution with onClick, but this method is called after onFocus, so, when onFocus 'opens' the calendar, the onClick method is called and immediately closes it. This could be fixed passing the open/close logic only for the onClick method, but in this case, the calendar would no longer appear when only the onFocus method is called (when the user uses 'tab', for example).

Reply:
Can the `onClick` call `setOpen(true)` instead of `setOpen(!open)`?

Reply:
This could be a solution, but this "toggle" behavior would be lost. I thought of put some logic with React states to achieve the same result, but without using the `onMouseDown` event. What you guys think about it?

Reply:
I'm wondering if we need the toggle behavior. Why would clicking on the input indicate that the user wants the Calendar Drop to be closed?

Reply:
I thought it was a expected feature, so I tried make it work correctly, but that would be a nice solution.